arm-iothub/2017-01-19/swagger/iothub.json

```diff
@@ -0,0 +1,1962 @@
+            "schema": {
+              "$ref": "#/definitions/IotHubDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "delete": {
+        "tags": [ "DELETE" ],
+        "summary": "Delete an IoT hub.",
+        "description": "Delete an IoT hub.",
+        "operationId": "IotHubResource_Delete",
+        "x-ms-long-running-operation": true,
+        "consumes": [ ],
+        "produces": [ "application/json", "text/json" ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The name of the resource group that contains the IoT hub.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "description": "The name of the IoT hub to delete.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "The Iot Hub resource provider always returns a 202 Accepted status code with valid Location and Retry-After headers. The resource provider also sets the Azure-AsyncOperation header with a URL that points to the operation resource for this operation. Subsequent GET attempts on the resource after a DELETE operation return a resource representation that indicates a transitional provisioning state (such as Terminating). To retrieve the status of the operation, a client can either poll the URL returned in the Location header after the Retry-After interval, get the IoT Hub service status directly, or query the operation resource.",
+            "schema": {
+              "$ref": "#/definitions/IotHubDescription"
+            }
+          },
+          "200": {
+            "description": "This is returned as a response to the status polling request for the delete operation. The body contains the resource representation that indicates a transitional provisioning state.",
+            "schema": {
+              "$ref": "#/definitions/IotHubDescription"
+            }
+          },
+          "204": {
+            "description": "Once the long running delete operation completes successfully, a 204 No Content status code is returned when the status polling request finds the Iot hub metadata in the service and the status of the delete operation is set to a completed state."
+          },
+          "404": {
```

Comment:
The operationresults API in our service was implemented a while ago when the recommendation in the ARM spec was to place this API underneath the original request. 

https://<endpoint>/subscriptions/{subscriptionId}/{Original-URL-Fo-Request}/operationresults/{operationId}&api-version={api-version}

Due to this at the end of the long running delete ARM will throw a 404 because the original resource no longer exists. The new ARM recommendation is to move the operationresults API to 
underneath the subscription instead of the resource

https://<endpoint>/subscriptions/{subscriptionId}/ providers/{namespace}/locations/westus/operationresults/{operationId}&api-version={api-version}

However we haven't implemented this yet. Until then 404 will be a valid response for delete for us and we need this to be specified here to ensure good behavior for the autorest generated SDKs